src/idl_gen_cpp.cpp

```diff
@@ -233,6 +233,27 @@ class CppGenerator : public BaseGenerator {
     return keywords_.find(name) == keywords_.end() ? name : name + "_";
   }
 
+  std::string Name(const FieldDef &field) const {
+    // the union type field suffix is immutable
+    static size_t union_suffix_len = strlen(UnionTypeFieldSuffix());
+    // early return if no case transformation required
+    if (opts_.cpp_object_api_field_case == IDLOptions::Case_Unchanged)
+      return EscapeKeyword(field.name);
+    std::string name = field.name;
+    if (field.value.type.base_type == BASE_TYPE_UTYPE &&
```

Comment:
The condition `field.value.type.base_type == BASE_TYPE_UTYPE` is checked twice.
Better to introduce `const is_enum` flag.

Or introduce `std::string suffix` that will be unconditionally appended to the `name` on `return`.
In this case `static size_t union_suffix_len` can be removed.

Reply:
intention of the static len is to avoid repeatedly computing strlen of an immutable string and avoiding heap allocation in this relatively hot path; meanwhile your suggestion of a local bool (not is_enum, more is_union_type) is in next commit

Reply:
resolved in latest commit